tests/test_warm_reboot.py

```diff
@@ -780,3 +789,687 @@ def test_swss_port_state_syncup(dvs, testlog):
+
+    elif action == 'CREATE':
+        pattern = "c\|" + ".*" + table + ".*" + entry
+
+    elif action == 'DELETE':
+        pattern = "r\|" + ".*" + table + ".*" + entry
+
+    (exitcode, num) = dvs.runcmd(['sh', '-c', "awk \'/%s/,ENDFILE {print;}\' /var/log/swss/sairedis.rec | egrep \"%s\" | wc -l" % (marker, pattern)])
+
+    assert num.strip() == "1"
+
+
+def obtain_syslog_changes(dvs):
+
+    (exitcode, num) = dvs.runcmd(['sh', '-c',
+                                  'egrep "reconciliation:" /var/log/syslog | grep -v "no changes" | wc -l'])
+    return num.strip()
+
+
+#
+# Obtain the total number of changes observed in swss.rec for a particular table.
+#
+def obtain_swss_changes(dvs, marker, table):
+    pattern = table + "_TABLE"
+
+    (exitcode, num) = dvs.runcmd(['sh', '-c', "awk \'/%s/,ENDFILE {print;}\' /var/log/swss/swss.rec | egrep \"%s\" | wc -l" % (marker, pattern)])
+
+    return num.strip()
+
+
+#
+# Obtain the total number of changes observed in sairedis.rec for a particular table.
+#
+def obtain_sairedis_changes(dvs, marker, table):
+    pattern = "TYPE_" + table + "_ENTRY"
+
+    (exitcode, num) = dvs.runcmd(['sh', '-c', "awk \'/%s/,ENDFILE {print;}\' /var/log/swss/sairedis.rec | egrep \"%s\" | wc -l" % (marker, pattern)])
+
+    return num.strip()
+
+
+def check_restart_timer(dvs, timer_value):
+    (exitcode, num) = dvs.runcmd(['sh', '-c',
+                                  'grep "Getting warmStartTimer" /var/log/syslog | rev | cut -d ' ' -f 1 | rev'])
+    assert num.strip() == timer_value
+
+
+# Temporary instruction to activate warm_restart. To be deleted once equivalent CLI
+# function is pushed to sonic-utils.
+def enable_warmrestart(dvs, db, app_name):
+    create_entry_tbl(
+        db,
+        swsscommon.CFG_WARM_RESTART_TABLE_NAME, app_name,
+        [
+            ("enable", "true"),
+        ]
+    )
+
+
+################################################################################
+#
+# Routing warm-restart testcases
+#
+################################################################################
+
+def test_routing_WarmRestart(dvs, testlog):
+
+    appl_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+    conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+    state_db = swsscommon.DBConnector(swsscommon.STATE_DB, dvs.redis_sock, 0)
+
+    # Restart-timer to utilize during the following testcases
+    restart_timer = 10
+
+
+    #############################################################################
+    #
+    # Baseline configuration
+    #
+    #############################################################################
+
+
+    # Defining create neighbor entries (4 ipv4 and 4 ip6, two each on each interface) in linux kernel
+    intfs = ["Ethernet0", "Ethernet4", "Ethernet8"]
+
+    # Enable ipv6 on docker
+    dvs.runcmd("sysctl net.ipv6.conf.all.disable_ipv6=0")
+
+    dvs.runcmd("ifconfig {} 111.0.0.1/24 up".format(intfs[0]))
```

Comment:
Do not use ifconfig. It's been deprecated and it's the source of many problems. Unit tests shouldn't break when sonic decides to remove net-tools.

Reply:
You shouldn't introduce and knowingly use functionality that has been deprecated. That's the whole point of the review. Besides, you are using ip for IPv6 so I'd rather you don't use ifconfig and you use ip  for IPv4 to match what your tests are doing for IPv6.

Reply:
done